projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -4502,9 +4502,8 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
      * @hidden
      */
     protected reinitPinStates() {
-        if (this.hasColumnGroups) {
-            this._pinnedColumns = this.columnList.filter((c) => c.pinned);
-        }
+        this._pinnedColumns = this.columnList.filter((c) => c.pinned);
```

Comment:
There are also some failing tests possibly due to the order of the pinned columns not being retained when the state is reinitialized (specifically when columns are moved in the pinned area).
This can probably be fixed by sorting the columns by their original order in the _pinnedColumns collection before setting it.